clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -71,18 +71,18 @@
     private static final String REPLICAS_KEY_NAME = "replicas";
     private static final String ISR_KEY_NAME = "isr";
 
-    private final Cluster cluster;
-    private final Map<String, Errors> errors;
+    private final Collection<Node> brokers;
+    private final List<TopicMetadata> topicMetadata;
 
-    /**
-     * Constructor for MetadataResponse where there are errors for some of the topics,
-     * error data take precedence over cluster information for particular topic
-     */
-    public MetadataResponse(Cluster cluster, Map<String, Errors> errors) {
+
+    public MetadataResponse(List<Node> brokers, List<TopicMetadata> topicMetadata) {
```

Comment:
This question also came up in the rack-aware PR. Will let you know if I get clarification from Jun on that one.